### ISSUE TITLE
should we add `self.text_encoder.eval()` ?

### ISSUE THREAD

**Issue by linhaojia13** [yanmin-wu/EDA#9]
To freeze the text encoder, you set the `param.requires_grad` as `False`:
```
self.tokenizer = RobertaTokenizerFast.from_pretrained(t_type, local_files_only=True)
self.text_encoder = RobertaModel.from_pretrained(t_type, local_files_only=True)
for param in self.text_encoder.parameters():
    param.requires_grad = False
```
However, to make sure the Dropout layer in the `RobertaModel` work as in evaluation (do not drop out neurons randomly), we need add ` self.text_encoder.eval()` at the end of the codes above:
```
self.tokenizer = RobertaTokenizerFast.from_pretrained(t_type, local_files_only=True)
self.text_encoder = RobertaModel.from_pretrained(t_type, local_files_only=True)
for param in self.text_encoder.parameters():
    param.requires_grad = False
self.text_encoder.eval()
```

**Comment by yanmin-wu** [comment]
Yes, your intuition is correct. However, you can check the `self.text_encoder.training` attribute (a boolean type) and find that the model is already in `eval()` mode. 
This is because in Hugging Face models, `the model is set in evaluation mode by default using model.eval() (Dropout modules are deactivated). To train the model, you should first set it back in training mode with model.train()`, please refer to: https://huggingface.co/docs/transformers/main_classes/model.

**Comment by linhaojia13** [comment]
Thank you very much!
